indra/newview/llviewerwindowlistener.cpp

```diff
@@ -94,14 +88,36 @@ void LLViewerWindowListener::saveSnapshot(const LLSD& event) const
         TypeMap::const_iterator found = types.find(event["type"]);
         if (found == types.end())
         {
-            LL_ERRS("LLViewerWindowListener") << "LLViewerWindowListener::saveSnapshot(): "
-                                              << "unrecognized type " << event["type"] << LL_ENDL;
-        return;
+            return response.error(stringize("Unrecognized type ", std::quoted(event["type"].asString()), " [\"COLOR\"] or [\"DEPTH\"] is expected."));
         }
         type = found->second;
     }
-    bool ok = mViewerWindow->saveSnapshot(event["filename"], width, height, showui, showhud, rebuild, type);
-    sendReply(LLSDMap("ok", ok), event);
+
+    std::string filename(event["filename"]);
+    if (filename.empty())
+    {
+        return response.error(stringize("File path is empty."));
+    }
+
+    LLSnapshotModel::ESnapshotFormat format(LLSnapshotModel::SNAPSHOT_FORMAT_BMP);
+    std::string ext = gDirUtilp->getExtension(filename);
+    if (ext.empty())
+    {
+        filename.append(".bmp");
+    }
+    else if (ext == "png")
+    {
+        format = LLSnapshotModel::SNAPSHOT_FORMAT_PNG;
+    }
+    else if (ext == "jpeg" || ext == "jpg")
+    {
+        format = LLSnapshotModel::SNAPSHOT_FORMAT_JPEG;
+    }
+    else if (ext != "bmp")
+    {
+        return response.error(stringize("Unrecognized format. [\"png\"], [\"jpeg\"] or [\"bmp\"] is expected."));
+    }
+    response["result"] = mViewerWindow->saveSnapshot(filename, width, height, showui, showhud, rebuild, type, format);
```

Comment:
Line 50 describes the original return key `"ok"`. I don't mind your changing it to `"result"` -- I don't expect anyone is still running original VITA tests -- but please update the documentation string if you do.